posydon/interpolation/profile_interpolation.py

```diff
@@ -492,20 +545,41 @@ def __init__(self,initial,h_profiles,he_profiles,star_state,
             self.valid_sort_ind[name] = np.where(self.valid_star_state==name)[0]
         
         # create and train models for profile boundaries
-        self.bounds_models, self.loss_history = self.learn_bounds(training_epochs,training_patience)
+        self.bounds_models, self.loss_history = self.learn_bounds(training_epochs,
+                                                                  training_patience,
+                                                                  depth=depth,
+                                                                  width=width,
+                                                                  lr=lr)
         
-    def learn_bounds(self,training_epochs,training_patience):
+    def learn_bounds(self,training_epochs,training_patience,depth,width,lr):
         """Creates and trains NNs to predict boundary points for each star 1 state.
         Args:
             training_epochs (int) : number of epochs used to train neural networks
             training_patience (int) : patience parameter for callback in neural networks
+            depth (int) : depth of neural network
+            width (int) : width of neural network
+            lr (float) : learning rate for neural network training
         Returns:
             b_models (array-like) : dictionary containing boundary models.
             loss_history (array-like) : training and validation loss histories
         """
         b_models = {}
         loss_history = {}
-        self.empty = [] # keep track of which star states have no data
+        
+        def calc_bevel_bounds(profs): 
+            # calculate first and last points in each profile with large increases
+            # i.e. boundaries of 'bevel' shape
+            nonflat=[] # ensures that training data only has the correct "non-flat" shape
+            bounds = []   # collect information on parameters for given H and He profile shapes
+            for i in range(len(profs)):
+                diff = np.where(profs[i][1:]-profs[i][:-1]>0.002)[0]
+                if len(diff>=2):
+                    bounds.append(np.array([diff[0]+1,diff[-1]+1])/200)
+                    nonflat.append(i)
+                else:
+                    bounds.append([np.nan,np.nan])
```

Comment:
Just for consistency: I'd suggest to either always append a numpy array or a list. line 577 vs. 580.

Reply:
done